src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -146,268 +69,68 @@ private void PopulateBackupMasker(bool generateCorrelatingIds)
                                                      this.regexEngine);
     }
 
-    public UInt32 PossibleMatches
+    public IEnumerable<Detection> DetectSecrets(string input)
     {
-        get
+        foreach (var hpDetection in highPerformanceSecretScanner.Scan(input))
         {
-            if (this.scan != IntPtr.Zero)
+            string found = input.Substring(hpDetection.Start, hpDetection.Length);
+            foreach (var pattern in this.signatureToPatternsMap[hpDetection.Signature])
             {
-                return identifiable_scan_match_count(this.scan);
+                Detection detection = ProcessDetection(hpDetection, found, pattern);
```

Comment:
>ProcessDetection [](http://example.com/codeflow?start=38&length=16)

So this name is super generic. You didn't create this problem. Maybe we could put some thought into it. `FinalizeDetection` might be better, this code calls into a refinement phase that itself might return null (in which case the detection is dropped entirely)